docs(release-notes): add 8.7.0 release notes (#5259)
canonical/snapcraft: docs/release-notes/snapcraft-8-7-0.rst

.. _release-8.7.0:

> Snapcraft 8.7.0 release notes

Author: ```suggestion
Snapcraft 8.7 release notes
```

=============================

03 March 2025

Learn about the new features, changes, and fixes introduced in Snapcraft 8.7.0.

Requirements and compatibility
------------------------------

See :ref:`System requirements<system-requirements>` for information on the minimum
hardware and installed software.

What's new
----------

Snapcraft 8.7.0 brings the following features, integrations, and improvements.

Bash completion
~~~~~~~~~~~~~~~

We've updated the completion file for the Snapcraft snap, which provides command-line
completion of commands and options in Bash-compatible shells. Try it out by
typing ``snapcraft`` and pressing :kbd:`Tab` in your terminal.

Previously, the completion file was outdated or incorrect because it had to be
manually updated. Now it's generated dynamically, which means it will always be

> up-to-date with the current commands and options.

Author: Make the first sentence it's own paragraph and first.  The second sentence should be blended with the other paragraph.

Improved remote builder
~~~~~~~~~~~~~~~~~~~~~~~

Remote builds can use the ``--build-for`` option to filter entries in a
``architectures`` or ``platforms`` field in a project file.

Support for confdbs
~~~~~~~~~~~~~~~~~~~

Snapcraft now supports listing and editing ``confdbs`` with the commands

> ``list-confdbs`` and ``edit-confdbs``. This replaces the previous commands,
> ``list-registries`` and ``edit-registries``.

Author: ```suggestion
``list-confdbs`` and ``edit-confdbs``. Thes commands replaces the previous commands,
``list-registries`` and ``edit-registries`` respectively.
```

Support for relocatable pkgconfig files
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Snapcraft now checks if pkgconfig (``.pc``) files are relocatable. If the file is
relocatable, Snapcraft will not modify the file's ``prefix`` field.

> Docs

Author: ```suggestion
Doc improvements
```

~~~~

> The following how-to guides have been migrated:

Author: ```suggestion
The following how-to guides have been integrated:
```

* :ref:`Enable experimental extensions <enable-experimental-extensions>`
* :ref:`List extensions <list-extensions>`
* :ref:`Use an extension <use-an-extension>`
* :ref:`Use the env-injector extension <use-the-env-injector-extension>`
* :ref:`Use the GNOME extension <use-the-gnome-extension>`

The following reference docs have been migrated:

> * :ref:`ROS 1 extension <ros-1-extension>`

Author: Copy order of how they are listed in the index

* :ref:`ROS 1 content extensions <ros-1-content-extensions>`
* :ref:`ROS 2 extensions <ros-2-foxy-extension>`
* :ref:`ROS 2 content extensions <ros-2-content-extensions>`
* :ref:`Flutter extension <flutter-extension>`
* :ref:`KDE neon extensions <kde-neon-extensions>`
* :ref:`env-injector extension <env-injector-extension>`
* :ref:`GNOME extension <gnome-extension>`

Backwards-incompatible changes
------------------------------

> ``--platform`` option for remote-build

Author: ```suggestion
Removal of ``--platform`` option for remote-build
```

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The ``--platform`` option has been dropped for the ``remote-build`` command.

This option was dropped because it doesn't provide predictable results for remote
builds, due to Launchpad's handling of the ``platforms`` field in project files.

``--build-for`` is the recommended alternative until Launchpad has comprehensive
support for platforms.

Known issues
------------

The following issues were reported and are scheduled to be fixed in upcoming
patch releases.

See individual issue links for any mitigations.

- `#4996`_ Remote build gives an unfriendly error when attempting to cross-compile.
- `#5258`_ The Flutter plugin fails to install Flutter for ``core22`` and ``core24``
  snaps.

Fixed bugs and issues
---------------------

The following issues have been resolved in Snapcraft 8.7.0:

- `#5250`_ Resources path for ``QtWebEngineProcess`` wasn't exported for snaps
  using the KDE Neon 6 extension.

> - `charmcraft#2069`_ The same build environment may be re-used for platforms with

Author: Replace with https://github.com/canonical/craft-application/issues/600
